ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/SnapWorldStateDownloader.java

```diff
@@ -183,16 +183,6 @@ public CompletableFuture<Void> run(
       } else {
         // start from scratch
         worldStateStorageCoordinator.clear();
-        // we have to upgrade to full flat db mode if we are in bonsai mode
-        if (snapSyncConfiguration.isFlatDbHealingEnabled()) {
```

Comment:
we don't have to downgradeToPartialFlatDbMode  if a user disable the full flat flag ? as the default is full now

Reply:
that is a good point.  I hadn't considered that anybody would want to run in a partial mode

Reply:
We can prevent user from doing that. It would not be logic